tutorials/iot/graphical-snaps/graphical-snaps-xwayland.md

```diff
@@ -0,0 +1,417 @@
+git push -u origin master
+```
+
+Now [setup your build on Launchpad](https://docs.snapcraft.io/build-snaps/ci-integration). *Note that you will need to use the same snap name in the store as in name in `name:`, so chose something unique to make your life easier.*
+
+Don't bother with publishing to the store (yet) you can download the snap and deploy it as follows:
+
+On your desktop go to the snaps webpage (e.g. https://code.launchpad.net/~/+snaps), find the build for your device architecture and download it and copy to your device. For example:
+```bash
+wget https://code.launchpad.net/~alan-griffiths/+snap/iot-example-graphical-xwayland-snap/+build/228782/+files/iot-example-graphical-xwayland-snap_0.1_arm64.snap
+scp iot-example-graphical-xwayland-snap_0.1_arm64.snap alan-griffiths@192.168.1.159:~
+```
+On your ssh session to your device:
+```bash
+snap install --dangerous iot-example-graphical-xwayland-snap_0.1_arm64.snap --devmode
+snap connect iot-example-graphical-xwayland-snap:wayland-socket-dir mir-kiosk:wayland-socket-dir
+sudo snap run iot-example-graphical-xwayland-snap.glxgears-kiosk
+```
+
+On your device, you should see the same graphical animation you saw earlier (and statistics printed to your console).
+
+## Confining Xwayland
+duration: 10:00
+
+One thing still needs to be resolved: this snap is installed with the `--devmode` option.
+
+To use fully confined snaps which use Xwayland internally, you will need a build of snapd that allows Xwayland to work in a confined snap. At the time of writing snapd "edge" already has [the necessary patch](https://github.com/snapcore/snapd/pull/4545). In the fullness of time this feature (to allow X11 slot implementations) will be be available on "stable", but for now we need to switch "core" to edge.
+
+In your ssh session to your device:
+```bash
+snap refresh core --edge
+```
+This will take a while and cause your device to reboot. So leave that running and reconnect ssh once we've updated the snap.
+
+To update the snap to use the X11 interface we also need to update the "snap/snapcraft.yaml" file to add the X11 plug & slot. The full file again:
+```yaml
+name: iot-example-graphical-xwayland-snap
+version: 0.1
+summary: glxgears on XWayland
+description: |
+   glxgears on XWayland
+confinement: strict
+grade: devel
+
+apps:
+ glxgears-kiosk:
+   command: xwayland-kiosk-launch glxgears
+   environment:
+     XWAYLAND_FULLSCREEN_WINDOW_HINT: title="glxgears"
+     WAYLAND_SOCKET_DIR: $SNAP_DATA/wayland
+   plugs:
+     - opengl
+     - wayland
+     - x11-plug
+   slots:
+     - x11
+
+plugs:
+  x11-plug: # because cannot have identical plug/slot name in same yaml.
+    interface: x11
+
+parts:
+ glxgears:
+   plugin: nil
+   after: [ xwayland-kiosk-helper ]
+   stage-packages:
+     - mesa-utils
+
+plugs:
+  wayland-socket-dir:
+    content: wayland-socket-dir
+    interface: content
+    target: $SNAP_DATA/wayland
+    default-provider: mir-kiosk
```

Comment:
I just tested this file as-is, and installing it I got 
2018-06-06T13:15:12Z INFO snap "iot-example-graphical-xwayland-snap" has bad plugs or slots: x11-plug (unknown interface "x11-plug")

Seems snapcraft doesn't deal well with having two "plugs" sections. If I combined them into one, that warning was removed.